### ISSUE TITLE
Conditional Templates are not shown correctly on init in SSR

### ISSUE THREAD

**Issue by web-dave** [angular/angular#58712]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

Yes

### Description

In a Template where I show conditional Data with @if/@else both dom nodes are shown.
This behaviour happens only if i enable ssr

In my stackblitz you can see this behaviour happens on if/else and if/if

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/github-k4ywtw?file=src%2Fapp%2Fapp.component.ts

### Please provide the exception or error you saw

```true
On start my app component has the value isLoading which is set to true.

In my template i want to show Dom regarding to isLoading.
At first see `loading` and `done`
When isLoading is set to false, UI updates to `Done` only

I would expect to see loading until isLoading was set to false.
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 18.2.12
Node: 18.19.1
Package Manager: npm 10.2.4
OS: darwin arm64

Angular: 18.2.12
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router, ssr

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1802.12
@angular-devkit/build-angular   18.2.12
@angular-devkit/core            18.2.12
@angular-devkit/schematics      18.2.12
@schematics/angular             18.2.12
rxjs                            7.8.1
typescript                      5.5.4
zone.js                         0.14.10
```

### Anything else?

This issue was reported at workshops_de discord by @muuvmuuv and kev

**Comment by JoostK** [comment]
I believe this is a duplicate of #57334.

**Comment by JeanMeche** [comment]
Yes, it's the same issue I'm closing as duplicate. Let's continue over there. Thank you for the repro Dave !

**Comment by thePunderWoman** [comment]
Re-opening this to just provide a comment here on the exploration.

So I spent some time with your reproduction here and determined there that the ngOnInit timeout was happening both on the server and the client, which changes the serialized hydration annotations. Essentially the server annotates _post_ timer execution, and this results in hydration being unable to match the content on the client because hydration happens before the timer has executed.

By restricting the timeout to just on the client, the problem goes away, and this is because the server serialization is stable and outputs the right things. You can see that in my fork of the repro here: https://stackblitz.com/edit/github-k4ywtw-hzzzcaca?file=src%2Fapp%2Fapp.component.ts. If you remove the isBrowser check, the content will duplicate.
